### ISSUE TITLE
Wrong indentation with long patterns and break-cases=all

### ISSUE THREAD

**Issue by bbc2** [ocaml-ppx/ocamlformat#766]
I get the following output:

```ocaml
let _ =
  match foo with
  | Reasonably_long_constructor_name (Other_reasonably_long_constructor_name _)
->
      0
```

with ocamlformat 0.9 and the following configuration:

```
break-cases=all
```

It only happens when the line with the pattern is 77, 78 or 79 characters long, the default margin being 80 characters.  Shorter lines get the arrow at the end and longer lines are wrapped.

With default `break-cases`, the arrow is correctly indented in all cases.

**Comment by gpetiot** [comment]
I get the following outout on 0.9 (and on master):
```ocaml
let _ =            
  match foo with
  | Reasonably_long_constructor_name
      (Other_reasonably_long_constructor_name _) ->
      0
```
do you use other options in addition to `break-cases`?

**Comment by bbc2** [comment]
No other configuration:

```
> tree -a
.
├── .ocamlformat
└── test.ml

0 directories, 2 files
```
```
> ocamlformat --version
0.9
```
```
> ocamlformat --print-config
profile=ocamlformat
quiet=false
max-iters=10
comment-check=true
wrap-fun-args=true
wrap-comments=false
type-decl=compact
space-around-collection-expressions=false
single-case=compact
sequence-style=separator
parse-docstrings=false
parens-tuple-patterns=multi-line-only
parens-tuple=always
parens-ite=false
ocp-indent-compat=false
module-item-spacing=sparse
margin=80
let-open=preserve
let-binding-spacing=compact
let-and=compact
leading-nested-match-parens=false
infix-precedence=indent
indicate-nested-or-patterns=space
indicate-multiline-delimiters=true
if-then-else=compact
field-space=tight
extension-sugar=preserve
escape-strings=preserve
escape-chars=preserve
doc-comments-tag-only=default
doc-comments-padding=2
doc-comments=after
disable=false
cases-exp-indent=4
break-struct=force
break-string-literals=wrap
break-sequences=false
break-separators=before
break-infix-before-func=true
break-infix=wrap
break-fun-decl=wrap
break-collection-expressions=fit-or-vertical
break-cases=all (file .ocamlformat:1)
```

Same on master:

```
> ~/code/ocamlformat/_build/install/default/bin/ocamlformat --version
c8f7af0
```

**Comment by gpetiot** [comment]
It is fixed with the PR #762 :
```ocaml
let _ =
  match foo with
  | Reasonably_long_constructor_name (Other_reasonably_long_constructor_name _)
    ->
      0
```

**Comment by gpetiot** [comment]
Fixed with #762
